server/tls/tlscertificate_retriever_windows.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Microsoft. All rights reserved.
+
+package tls
+
+import (
+	"crypto"
+	"crypto/x509"
+	"fmt"
+	certtostore "github.com/Azure/azure-container-networking/server/tls/customCertToStore"
+	"golang.org/x/sys/windows"
+)
+
+type windowsTlsCertificateRetriever struct{
+	certStore *certtostore.WinCertStore
+	certContext *windows.CertContext
+	settings 	TlsSettings
+}
+// Get certificate reads from the windows cert store
+// it depends on the TlsCertificateSubjectName being set
+// in the server settings to retrieve the cert
+func (wtls *windowsTlsCertificateRetriever) GetCertificate() (*x509.Certificate, error){
+
+	if wtls.settings.TlsCertificateSubjectName == ""{
+		return nil, fmt.Errorf("Certificate subject name is not empty in the settings")
```

Comment:
fix the error - is empty

Reply:
I believe these checks should be done before starting the CNS service. I have a comment for this somewhere in main.go I believe

Reply:
There are two cases: 
1) Where we retrieve by subject name for windows from the windows cert store
2) Where we retrieve by file path on linux

The case where subject name is required is in this file windows file and in other it is not required, as such its the logical place to have this check.